### PR TITLE
support parsing Postgre sql `DROP CAST`

### LINKED ISSUE #15742: [OpenForce 2022] support parsing Postgre sql `DROP CAST`
Hi community,

This issue is for [OpenForce 2022](https://github.com/apache/shardingsphere/discussions/12683#discussioncomment-2139441)

## Background
ShardingSphere parser engine helps users parse a SQL to get the AST (Abstract Syntax Tree) and visit this tree to get SQLStatement (Java Object). At present, this parser engine can handle SQLs for `MySQL`, `PostgreSQL`, `SQLServer`, `openGauss` and `Oracle`, which means we have to understand different database dialect SQLs.
More details: 
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

## Task
This issue is to solve some of the unsupported postgres sqls  in [this file](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/unsupported/unsupported.xml) .

- [ ] DROP CAST

You can learn more details about this SQL [here](https://www.postgresql.org/docs/current/sql-commands.html).
- You may need to try to get why it's not supported.(antlr4 grammar? or not implement visit method) You can use [antlr4 plugins](https://plugins.jetbrains.com/plugin/7358-antlr-v4) to help you to analyze. You may need to visit [an official doc](https://www.postgresql.org/docs/current/sql-commands.html) to check the grammar.
- After you fix it, remember to add a new corresponding SQL case in [SQL Cases](https://github.com/apache/shardingsphere/tree/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported) and expected parsed result in [Expected Statment XML](https://github.com/apache/shardingsphere/tree/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/case).
- Run [SQLParserParameterizedTest](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/SQLParserParameterizedTest.java) and [UnsupportedSQLParserParameterizedTest](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/UnsupportedSQLParserParameterizedTest.java) to make sure no exceptions.
Notice, these issues can be a good example.
[support alter foreign table for pg/og ](https://github.com/apache/shardingsphere/pull/14944)
[support alter materialized view for pg/og.](https://github.com/apache/shardingsphere/pull/15331)

## Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr g4 file
3. Be familiar with Postgres SQLs
## Targets files
1. Postgres SQLs g4 file: https://github.com/apache/shardingsphere/blob/master/shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/org/apache/shardingsphere/sql/parser/autogen/PostgreSQLStatement.g4

### PR DESCRIPTION
**Pull request by everly-gif**
Fixes #15742 .

Changes proposed in this pull request:
- Adds `DROP CAST`
- Adds tests

### CODE REVIEW

**Comment by everly-gif** [comment]
Some of the tests are failing not sure why. Would appreciate guidance.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/ddl/drop-cast.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<sql-parser-test-cases>
+    <drop-cast sql-case-id="drop_cast_if_exists_case1" />
```
[the remark this answers is not in the document]
Comment (everly-gif) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/ddl/drop-cast.xml:20]: Yes, agreed have changed the order

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-cast.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+ <?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (everly-gif) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-cast.xml]: Thank you @ThanoshanMV . I have pushed the changes and fixed the conflicts.

**Review by ThanoshanMV — DISMISSED** [review]

**Comment by ThanoshanMV** [comment]
Thank you @everly-gif for your contribution.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/ddl/drop-cast.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <drop-cast sql-case-id="drop_cast_if_exists_case2" />
+    <drop-cast sql-case-id="drop_cast_if_exists_case3" />
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/ddl/drop-cast.xml:22]: Hi @everly-gif, can you rename these `sql-case-id` to meaningful names?

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-cast.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<sql-cases>
+    <sql-case id="drop_cast_if_exists_case1" value="DROP CAST IF EXISTS (INTEGER AS no_such_schema.bar);" db-types="PostgreSQL"/>
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-cast.xml:20]: @everly-gif Same problem.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-cast.xml (hunk 4 above)
Comment (everly-gif) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-cast.xml]: changed

**Review by strongduanmu — APPROVED** [review]
@everly-gif Good job, merged.

**Pull request merged**